packages/components/src/font-size-picker/index.tsx

```diff
@@ -130,56 +183,60 @@ const UnforwardedFontSizePicker = (
 							</HeaderHint>
 						) }
 					</HeaderLabel>
-					{ ! disableCustomFontSizes && (
+					{ /* Show toggle button only when both picker modes are enabled */ }
+					{ computedPickerMode === 'both' && (
 						<HeaderToggle
 							label={
-								showCustomValueControl
+								currentPickerType === 'custom'
 									? __( 'Use size preset' )
 									: __( 'Set custom size' )
 							}
 							icon={ settings }
 							onClick={ () => {
-								setShowCustomValueControl(
-									! showCustomValueControl
+								setCurrentPickerType(
+									getPickerType(
+										currentPickerType === 'custom'
+											? 'predefined'
+											: 'custom',
+										isCustomValue,
+										fontSizes
+									)
 								);
 							} }
-							isPressed={ showCustomValueControl }
+							isPressed={ currentPickerType === 'custom' }
 							size="small"
 						/>
 					) }
 				</Header>
 			</Spacer>
 			<div>
-				{ !! fontSizes.length &&
-					shouldUseSelectControl &&
-					! showCustomValueControl && (
```

Comment:
The logic for what UI should be shown to the user has been consolidated and moved in the `getPickerType` utility.

Also, the `!! fontSizes.length` checked seemed redundant, since `shouldUseSelectControl` already assumes that there are more than 5 font sizes